app/src/main/java/com/elmeradrianv/shesafe/fragments/MapFragment.java

```diff
@@ -134,15 +184,85 @@ public void onLocationChanged(Location location) {
         if (location == null) {
             return;
         }
+        double longitudePFL = location.getLongitude() - currentLocation.getLongitude();
+        double latitudePFL = location.getLatitude() - currentLocation.getLatitude();
         currentLocation = location;
-        displayLocation();
+        LatLng speedVector = new LatLng(latitudePFL, longitudePFL);
+        LatLng actualLocation = new LatLng(
+                currentLocation.getLatitude(),
+                currentLocation.getLongitude());
+    }
+
+    private double positiveRemainder(double divisor, double dividend) {
+        double quotient = Math.floor(dividend / divisor);
+        return dividend - divisor * quotient;
+    }
+
+    private HashMap<Integer, ParsePolygon> getActualGridSquare() {
+        HashMap<Integer, ArrayList<ParseGeoPoint>> grid = new HashMap<>();
+        //squareSize is a "debug number and it will change depends of the speedVector"
+        double squareSize = 0.005;
+        int numSquareSizeGrid = 3;
+        double cornerLatitudeCC = currentLocation.getLatitude() - positiveRemainder(squareSize, currentLocation.getLatitude());
+        double cornerLongitudeCC = currentLocation.getLongitude() - positiveRemainder(squareSize, currentLocation.getLongitude());
+        //saving the number of points in the grid
+        LatLng[][] gridCorners = new LatLng[numSquareSizeGrid + 1][numSquareSizeGrid + 1];
+        for (int i = 0; i < gridCorners.length; i++) {
+            for (int j = 0; j < gridCorners[0].length; j++) {
+                gridCorners[i][j] = new LatLng(
+                        cornerLatitudeCC - (1 - i) * squareSize,
+                        cornerLongitudeCC - (1 - j) * squareSize
+                );
+            }
+        }
+        //k is nine because the grid has 9 squares
+        for (int k = 0; k < 9; k++) {
+            grid.put(k, new ArrayList<>());
+            PolygonOptions polygonOptions = new PolygonOptions();
+            for (int i = k % numSquareSizeGrid; i < k % numSquareSizeGrid + 2; i++) {
```

Comment:
This is getting past the point where i,j,k will work.

I'd recommend something like this (names might not be right but adjust as needed):
int gridColumn = k % numSquareSizeGrid;
int gridRow = k / numSquareSizeGrid; 
for (int column = gridColumn; column < gridColumn + 2; i++)

etc...

Any time you're repeating the same calculation just name it and use the variable. It'll help with readability.